packages/build/src/core/build.js

```diff
@@ -0,0 +1,554 @@
+  errorParams,
+  logs,
+  debug,
+  systemLog,
+  verbose,
+  sendStatus,
+  saveConfig,
+  timers,
+  testOpts,
+  buildbotServerSocket,
+  constants,
+  featureFlags,
+  timeline,
+  devCommand,
+}) {
+  const { pluginsOptions: pluginsOptionsA, timers: timersA } = await getPluginsOptions({
+    pluginsOptions,
+    netlifyConfig,
+    siteInfo,
+    buildDir,
+    nodePath,
+    packageJson,
+    userNodeVersion,
+    mode,
+    api,
+    logs,
+    debug,
+    sendStatus,
+    timers,
+    testOpts,
+    featureFlags,
+  })
+  // eslint-disable-next-line fp/no-mutation, no-param-reassign
+  errorParams.pluginsOptions = pluginsOptionsA
+
+  const { childProcesses, timers: timersB } = await startPlugins({
+    pluginsOptions: pluginsOptionsA,
+    buildDir,
+    childEnv,
+    logs,
+    debug,
+    timers: timersA,
+  })
+
+  try {
+    const {
+      stepsCount,
+      netlifyConfig: netlifyConfigA,
+      statuses,
+      failedPlugins,
+      timers: timersC,
+      configMutations,
+    } = await runBuild({
+      childProcesses,
+      pluginsOptions: pluginsOptionsA,
+      netlifyConfig,
+      configOpts,
+      packageJson,
+      configPath,
+      headersPath,
+      redirectsPath,
+      buildDir,
+      repositoryRoot,
+      nodePath,
+      childEnv,
+      context,
+      branch,
+      dry,
+      buildbotServerSocket,
+      constants,
+      mode,
+      api,
+      errorMonitor,
+      deployId,
+      errorParams,
+      logs,
+      debug,
+      systemLog,
+      verbose,
+      saveConfig,
+      timers: timersB,
+      testOpts,
+      featureFlags,
+      timeline,
+      devCommand,
+    })
+
+    await Promise.all([
+      warnOnMissingSideFiles({ buildDir, netlifyConfig: netlifyConfigA, logs }),
+      warnOnLingeringProcesses({ mode, logs, testOpts }),
+    ])
+
+    return {
+      stepsCount,
+      netlifyConfig: netlifyConfigA,
+      statuses,
+      pluginsOptions: pluginsOptionsA,
+      failedPlugins,
+      timers: timersC,
+      configMutations,
+    }
+  } finally {
+    stopPlugins(childProcesses)
+  }
+}
+
+// Load plugin main files, retrieve their event handlers then runs them,
+// together with the build command
+const runBuild = async function ({
+  childProcesses,
+  pluginsOptions,
+  netlifyConfig,
+  configOpts,
+  packageJson,
+  configPath,
+  headersPath,
+  redirectsPath,
+  buildDir,
+  repositoryRoot,
+  nodePath,
+  childEnv,
+  context,
+  branch,
+  dry,
+  buildbotServerSocket,
+  constants,
+  mode,
+  api,
+  errorMonitor,
+  deployId,
+  errorParams,
+  logs,
+  debug,
+  systemLog,
+  verbose,
+  saveConfig,
+  timers,
+  testOpts,
+  featureFlags,
+  timeline,
+  devCommand,
+}) {
+  const { pluginsSteps, timers: timersA } = await loadPlugins({
+    pluginsOptions,
+    childProcesses,
+    packageJson,
+    timers,
+    logs,
+    debug,
+    verbose,
+  })
+
+  const { steps, events } = timeline === 'dev' ? getDevSteps(devCommand, pluginsSteps) : getSteps(pluginsSteps)
```

Comment:
This is the main change here. We get the `timeline` and `devCommand` flags from upstream and choose a different set of steps based on the value of `timelime` (which defaults to `"build"`).